apps/web/pages/category/[slug].vue

```diff
@@ -38,8 +38,73 @@ const { data: categoryTree } = useCategoryTree();
 const { locale } = useI18n();
 const localePath = useLocalePath();
 
+const setCategoryCanonical = () => {
```

Comment:
this function has to go into a composable.

Reply:
i added it to useCanonical but it goes over the 15 line limit set in eslint? should i create a separate composable?

Reply:
it is more comprehensive to have use canonical and functions in there

Reply:
I would split the composable method in two. So that you extract the category pagination into its own function.